src/request/components/RequestDetailPanelRequest.tsx

```diff
@@ -72,11 +74,34 @@ export class Request extends React.Component<IRequestProps, {}> {
         );
     }
 
-    private renderBody(body: string) {
+    private renderBody(body: string, headers: { [key: string]: string }) {
+        const contentType = this.getContentTypeFromHeaders(headers);
+        const highlightClassName = this.getHighlightClassNameForContentType(contentType);
+
         return (
             <div className='tab-request-body'>
-                <Highlight className=''>{body}</Highlight>
+                <Highlight className={highlightClassName}>{body}</Highlight>
             </div>
         );
     }
+
+    private getContentTypeFromHeaders(headers: { [key: string]: string }): string {
+        let contentType = undefined;
+
+        _.forEach(headers, (value, key) => {
+            if (key.toLowerCase() === 'content-type') {
+                contentType = value;
+                
+                return false;
+            }
+        });
+
+        return contentType;
+    }
+
+    private getHighlightClassNameForContentType(contentType: string): string {
+        const category = requestConverter.getContentTypeCategory(contentType);
+
+        return (category && category.highlight) || '';
+    }
```

Comment:
Here is an interesting question... given that the class name is very view related, I wonder if this should come out of the selector. My guess is not.